frontend/src/components/VisaStatusOverviewChart.js

```diff
@@ -0,0 +1,229 @@
+import React, { useEffect, useMemo } from "react";
+import PropTypes from "prop-types";
+import ReactEcharts from "echarts-for-react";
+import { useSelector, useDispatch } from "react-redux";
+import { useTranslation } from "react-i18next";
+import {
+    makeFilterSelectorByVisaType,
+    makeDetailSelectorByVisaType,
+    makeOverviewSpanSelectorByVisaType,
+} from "../redux/selectors";
+import { fetchVisaStatusDetail } from "../redux/visastatusDetailSlice";
+import { getTimeFromUTC, getDateFromISOString } from "../utils/misc";
+import { useScreenXS } from "../hooks";
+
+const dataZoom = [
+    {
+        type: "slider",
+        height: 8,
+        bottom: 20,
+        borderColor: "transparent",
+        backgroundColor: "#e2e2e2",
+        handleIcon:
+            "M10.7,11.9H9.3c-4.9,0.3-8.8,4.4-8.8,9.4c0,5,3.9,9.1,8.8,9.4h1.3c4.9-0.3,8.8-4.4,8.8-9.4C19.5,16.3,15.6,12.2,10.7,11.9z M13.3,24.4H6.7v-1.2h6.6z M13.3,22H6.7v-1.2h6.6z M13.3,19.6H6.7v-1.2h6.6z",
+        handleSize: 20,
+        handleStyle: {
+            shadowBlur: 6,
+            shadowOffsetX: 1,
+            shadowOffsetY: 2,
+            shadowColor: "#aaa",
+        },
+    },
+    {
+        type: "inside",
+    },
+];
+
+const mergeDetailData = (rawData, vsFilter) => {
+    const delta = 60000;
+    const { timeRange, detail } = rawData;
+    if (timeRange.length === 0) return [[], []];
+    const [tsStart, tsEnd] = timeRange;
+    const writeTime = [];
+    for (let t = tsStart; t <= tsEnd; t += delta) writeTime.push(t);
+    const availDateLst = vsFilter.map(embassyCode => {
+        const availableDates = detail[embassyCode] || [];
+        let dataIndex = 0;
+        let current = null;
+        const avaDates = writeTime.map(t => {
+            if (dataIndex < availableDates.length && availableDates[dataIndex].writeTime === t) {
+                current = availableDates[dataIndex].availableDate;
+                dataIndex += 1;
+            }
+            return current === null ? null : current.join("/");
+        });
+        return { embassyCode, availableDates: avaDates };
+    });
+    return [writeTime, availDateLst];
+};
+
```

Comment:
```js
if (cond) doSomething();
```

and 

```js
for (cond) doSomething();
```

Are perfectly valid, but it makes the code unreadable. If I'm being opinionated I would suggest always use coding style like

```js
if (cond) {
    doSomething();
}

for (cond) {
    doSomething():
}
```

that comply with modern JS preferred coding style.